src/sage/categories/additive_magmas.py

```diff
@@ -766,7 +766,7 @@ class ElementMethods:
             #     return self == self.parent().zero()
 
             @abstract_method
-            def __bool__(self):
```

Comment:
/home/runner/work/sage/sage/src/sage/categories/additive_magmas.py:769:35 - warning: Function with declared return type "bool" must return value on all code paths
  Type "None" cannot be assigned to type "bool" (reportGeneralTypeIssues)